fs/snapshot.h

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+// Copyright (c) 2019-present, Western Digital Corporation
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+
+#include <string>
+#include <vector>
+#include "io_zenfs.h"
+#include "zbd_zenfs.h"
+
+namespace ROCKSDB_NAMESPACE {
+
+// These are three Snapshot classes to capture real-time information from ZenFS
+// for use by the upper layer algorithms. The three Snapshots will capture
+// information from Zone, ZoneFile, and ZoneExtent respectively. If you plan to
+// modify the variables of these three classes, please make sure that they have
+// a public interface for copying and that the interface of the Snapshot classes
+// are still logically correct after modification.
+
+class ZoneSnapshot {
+ private:
+  uint64_t start_;
+  uint64_t capacity_;
+  uint64_t max_capacity_;
+  uint64_t wp_;
+
+ public:
+  ZoneSnapshot(const Zone& zone)
+      : start_(zone.start_),
+        capacity_(zone.capacity_),
+        max_capacity_(zone.max_capacity_),
+        wp_(zone.wp_) {}
+
+ public:
```

Comment:
Should we remove duplicate `public` access specifier of a class/struct?

Reply:
Yes, it is somewhat redundant since only public functions are available.